staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1297,6 +1305,11 @@ func (c *Cacher) getCommonResourceVersionLockedFunc(ctx context.Context, parsedW
 // all events from the cache (full state) or not.
 func (c *Cacher) waitUntilWatchCacheFreshAndForceAllEvents(ctx context.Context, requestedWatchRV uint64, opts storage.ListOptions) (bool, error) {
 	if opts.SendInitialEvents != nil && *opts.SendInitialEvents {
+		if utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) && c.watchCache.notFresh(requestedWatchRV) {
```

Comment:
I think that this is okay because we will disallow streaming when the consistentListFromCache is disabled. (otherwise latency will be increased)